grpc/src/test/java/com/linecorp/armeria/client/grpc/GrpcClientBuilderTest.java

```diff
@@ -19,22 +19,72 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
+import java.io.InputStream;
+import java.util.concurrent.Executors;
+
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
 import com.linecorp.armeria.client.ClientBuilderParams;
 import com.linecorp.armeria.client.Clients;
 import com.linecorp.armeria.client.endpoint.EndpointGroup;
 import com.linecorp.armeria.common.SerializationFormat;
 import com.linecorp.armeria.common.grpc.GrpcSerializationFormats;
+import com.linecorp.armeria.internal.common.grpc.TestServiceImpl;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.grpc.GrpcService;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
 
 import io.grpc.CallOptions;
 import io.grpc.Channel;
 import io.grpc.ClientCall;
 import io.grpc.ClientInterceptor;
 import io.grpc.MethodDescriptor;
+import io.grpc.MethodDescriptor.Marshaller;
+import testing.grpc.Messages.SimpleRequest;
 import testing.grpc.TestServiceGrpc.TestServiceBlockingStub;
 
 class GrpcClientBuilderTest {
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.service(GrpcService.builder()
+                                  .addService(new TestServiceImpl(Executors.newSingleThreadScheduledExecutor()))
+                                  .build());
+        }
+    };
+
+    private static class CustomMarshallerInterceptor implements ClientInterceptor {
+        private int spiedMarshallerCallCnt;
+
+        CustomMarshallerInterceptor(int spiedMarshallCallCnt) {
+            this.spiedMarshallerCallCnt = spiedMarshallCallCnt;
+        }
+
+        int getSpiedMarshallerCallCnt() {
+            return spiedMarshallerCallCnt;
+        }
+
+        @Override
+        public <I, O> ClientCall<I, O> interceptCall(MethodDescriptor<I, O> method, CallOptions callOptions,
+                                                     Channel next) {
+            final MethodDescriptor<I, O> methodDescriptor = method.toBuilder().setRequestMarshaller(
```

Comment:
We can also test the server-side option with a custom `Marshaller`.
```java
var requestMarshaller = TestServiceGrpc.getUnaryCallMethod().getRequestMarshaller();
var testRequestMarshaller = new Marshaller<SimpleRequest>() {
    @Override
    public InputStream stream(SimpleRequest value) {
        return requestMarshaller.stream(value);
    }
    @Override
    public SimpleRequest parse(InputStream stream) {
        ...
    }
};

var responseMarshaller = TestServiceGrpc.getUnaryCallMethod().getResponseMarshaller();
var testResponseMarshaller = new Marshaller<SimpleRequest>() {
    @Override
    public InputStream stream(SimpleRequest value) {
        return responseMarshaller.stream(value);
    }
    @Override
    public SimpleRequest parse(InputStream stream) {
        ...
    }
};

// Override the default marshallers with custom ones.
final ServerServiceDefinition serviceDefinition = ServerInterceptors.useMarshalledMessages(
    testService.bindService(), testRequestMarshaller, testResponseMarshaller);

GrpcService.builder()
           .addService(serviceDefinition)
```

Reply:
Applied feedback. Thanks